### PR TITLE
Upgraded kruntime to beta3

### PR DESCRIPTION
**Pull request by Yantrio**
Everything seems to work for me in windows both in cli + visual studio

My only concern is the nulls passed around in UpdateBufferFilterFacts.cs, if anyone has input on what should go in there, it would be much appreciated :)

### CODE REVIEW

File: src/OmniSharp/Startup.cs [hunk 1]
```diff
@@ -49,7 +49,8 @@ public void ConfigureServices(IServiceCollection services, IApplicationLifetime
 
             services.Configure<MvcOptions>(opt =>
             {
-                opt.OutputFormatters.RemoveAll(r => r.Instance is XmlOutputFormatter);
+                //not needed anymore due to https://github.com/aspnet/Mvc/issues/1841 ?
+                //opt.OutputFormatters.RemoveAll(r => r.Instance is XmlOutputFormatter);
```
Comment (filipw) [inline src/OmniSharp/Startup.cs]: this can be deleted the XML formatter is now opt in not opt out

**Comment by nosami** [comment]
The build is failing because of how Travis is handling c# builds. It can't deal with beta3 yet.

We can either wait until they fix Travis, or use our own build script (similar to https://github.com/OmniSharp/omnisharp-server/blob/master/.travis.yml)

**Comment by nosami** [comment]
Created an issue on travis-ci

https://github.com/travis-ci/travis-ci/issues/3273

**Comment by jrieken** [comment]
@nosami Is this really Travis and not our build.sh script which still had references to ~/.kre in it?

**Comment by Yantrio** [comment]
well, our build.sh doesn't have a line 41, so that is a quick telltale sign that something is falling over that is out of our control.

**Comment by nosami** [comment]
Our code is cloned like this `git clone --depth=50 git://github.com/OmniSharp/omnisharp-roslyn.git OmniSharp/omnisharp-roslyn`

Our build.sh doesn't go into the home folder 

```bash
curl -sSL https://raw.githubusercontent.com/aspnet/Home/master/kvminstall.sh | sh && source ~/.kre/kvm/kvm.sh && kvm upgrade
Downloading kvm as script to '/home/travis/.k/kvm'
Downloading kvm from 'https://raw.githubusercontent.com/aspnet/Home/master/kvm.sh'
Appending source string to /home/travis/.bashrc
Appending source string to /home/travis/.zshrc
Type 'source /home/travis/.k/kvm/kvm.sh' to start using kvm
/home/travis/build.sh: line 41: /home/travis/.kre/kvm/kvm.sh: No such file or directory
```

**Comment by nosami** [comment]
Just in case there was any doubt, I reran the last good beta2 build https://travis-ci.org/OmniSharp/omnisharp-roslyn/builds/51087932

**Comment by nosami** [comment]
https://github.com/Yantrio/omnisharp-roslyn/blob/beta3/scripts/Omnisharp#L13

**Comment by nosami** [comment]
@jrieken do you happen to know which branch of XRE was used for beta3? We need to copy the DTH models.

**Comment by filipw** [comment]
there's a tag https://github.com/aspnet/XRE/tree/1.0.0-beta3

**Comment by Yantrio** [comment]
https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs#L272

This line is failing with a json conversion error :(

A first chance exception of type 'Newtonsoft.Json.JsonSerializationException' occurred in Newtonsoft.Json.dll

>Additional information: Cannot deserialize the current JSON object (e.g. {"name":"value"}) into type 'System.Collections.Generic.IEnumerable`1[System.Collections.Generic.KeyValuePair`2[System.String,Microsoft.CodeAnalysis.ReportDiagnostic]]' because the type requires a JSON array (e.g. [1,2,3]) to deserialize correctly.

>To fix this error either change the JSON to a JSON array (e.g. [1,2,3]) or change the deserialized type so that it is a normal .NET type (e.g. not a primitive type like integer, not a collection type like an array or List<T>) that can be deserialized from a JSON object. JsonObjectAttribute can also be added to the type to force it to deserialize from a JSON object.

>Path 'CompilationOptions.CompilationOptions.SpecificDiagnosticOptions.CS1702'.

and the message payload appears to be correct to me

```{{
  "Framework": {
    "FrameworkName": "Asp.Net,Version=v5.0",
    "FriendlyName": "ASP.NET 5.0",
    "ShortName": "aspnet50",
    "RedistListPath": "C:\\Program Files (x86)\\Reference Assemblies\\Microsoft\\Framework\\.NETFramework\\v4.5.1\\RedistList\\FrameworkList.xml"
  },
  "CompilationOptions": {
    "LanguageVersion": 6,
    "Defines": [
      "DEBUG",
      "TRACE",
      "ASPNET50"
    ],
    "CompilationOptions": {
      "AllowUnsafe": false,
      "Usings": [],
      "OutputKind": 2,
      "ModuleName": null,
      "ScriptClassName": "Script",
      "MainTypeName": null,
      "CryptoKeyFile": null,
      "CryptoKeyContainer": null,
      "DelaySign": null,
      "CheckOverflow": false,
      "Platform": 0,
      "OptimizationLevel": 0,
      "GeneralDiagnosticOption": 0,
      "WarningLevel": 4,
      "ConcurrentBuild": true,
      "SpecificDiagnosticOptions": {
        "CS1702": 5,
        "CS1705": 5
      },
      "MetadataReferenceResolver": null,
      "XmlReferenceResolver": null,
      "SourceReferenceResolver": null,
      "StrongNameProvider": null,
      "AssemblyIdentityComparer": {},
      "Errors": []
    }
  }
}}```

**Comment by Yantrio** [comment]
Ignore that, Fixed that error :D

**Comment by nosami** [comment]
:+1:

File: src/OmniSharp/AspNet5/OutgoingMessages/DependencyItem.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
```
Comment (nosami) [inline src/OmniSharp/AspNet5/OutgoingMessages/DependencyItem.cs]: not used

File: src/OmniSharp/AspNet5/OutgoingMessages/OutputsMessage.cs [hunk 3]
```diff
@@ -1,4 +1,5 @@
-﻿using System.Collections.Generic;
+﻿using System;
```
Comment (nosami) [inline src/OmniSharp/AspNet5/OutgoingMessages/OutputsMessage.cs]: not used

**Comment by nosami** [comment]
Awesome PR

File: build.cmd [hunk 4]
```diff
@@ -1,6 +1,6 @@
 @echo off
 
 set "KRE_NUGET_API_URL=https://www.nuget.org/api/v2"
-CALL kvm install 1.0.0-beta2
-CALL kvm use 1.0.0-beta2
-CALL kpm pack src/OmniSharp --no-source --out artifacts/build/omnisharp --runtime KRE-CLR-x86.1.0.0-beta2
+CALL kvm install 1.0.0-beta3
+CALL kvm use 1.0.0-beta3
+CALL kpm bundle src/OmniSharp --no-source --out artifacts/build/omnisharp --runtime KRE-CLR-win-x86.1.0.0-beta3
```
Comment (davidfowl) [inline build.cmd]: Shud be all lowercase here too:

```
kre-clr-win-x86
```

**Comment by davidfowl** [comment]
:clap:

**Pull request merged**